### ISSUE TITLE
Site object caching causes get_current_site to return wrong site

### ISSUE THREAD

**Issue by rjdp** [bernardopires/django-tenant-schemas#492]
I submitted a PR for this on Django(https://github.com/django/django/pull/8726), although it did not pass tests, it encapsulates the essence of a possible solution to this inconsistency.

**Comment by bernardopires** [comment]
Why would you want to use the sites framework together with this app? The tenants app should pretty much fulfill what sites was made for.

**Comment by rjdp** [comment]
not using sites framework, but on successive requests very often get_current_site returns wrong site due to caching by Id since all sites have same id i.e 1

**Comment by bernardopires** [comment]
`get_current_site`  is part of the sites framework. Why are you using get_current_site?

**Comment by rjdp** [comment]
Actually, I am not using the sites framework but I am using the sitemap framework(to generate sitemaps) which uses get_currentsite (from django.contrib.sites.shortcuts import get_current_site) see https://github.com/django/django/blob/master/django/contrib/sitemaps/views.py#L28 , https://github.com/django/django/blob/master/django/contrib/sitemaps/views.py#L53 . Thanks.

**Comment by bernardopires** [comment]
I actually don't see why sitemaps wouldn't work out of the box. You want one sitemap per tenant right? As long as the correct schema is set it should be able to fetch everything it needs.

**Comment by rjdp** [comment]
Indeed I am setting up correct schema via middleware but, if you see the definition of [`get_current_site`](https://github.com/django/django/blob/5411821e3b8d1427ee63a5914aed1088c04cc1ed/django/contrib/sites/shortcuts.py#L4)), `site` is being cached by `Id` across requests.

**Comment by bernardopires** [comment]
So it's just the URLs are wrong, but it's fetching everything correctly?

**Comment by rjdp** [comment]
Yes I am getting incorrect domain names as site objects some times belong to some other schema instead of current context's schema

**Comment by rjdp** [comment]
What do you think, Should this be fixed in the `tenant_schemas` or the `Dango` repository?

**Comment by bernardopires** [comment]
Django doesn't support schemas so there's no reason for it to be fixed on their part. You'll have however to customize the sitemap case for you specifically, by either creating a new sitemap app that works with tenant schemas or you modify your Django sitemap app to cache sites properly.

**Comment by rjdp** [comment]
Got it, thanks for your suggestion.
